### PR TITLE
[Tables] Use lists for batching

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_table_batch_async.py [hunk 1]
```diff
@@ -182,11 +182,11 @@ def _batch_create_entity(
...
         )
-        self._requests.append(request)
+        self.requests.append(request)
 
     _batch_create_entity.metadata = {"url": "/{table}"}  # type: ignore
 
-    def update_entity(
+    def update(
         self,
         entity,  # type: Union[TableEntity, Dict[str,str]]
```
Comment (seankane-msft) [inline sdk/tables/azure-data-tables/azure/data/tables/aio/_table_batch_async.py:191]: Can we update the type hints on entity here to use the `EntityType` defined previously?

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py [hunk 2]
```diff
@@ -37,19 +30,19 @@
...
         self,
-        account_url,  # type: str
-        table_name,  # type: str
-        credential=None,  # type: str
-        **kwargs  # type: Any
-    ):
-        # type: (...) -> None
+        account_url: str,
+        table_name: str,
+        credential: Optional[Union[AzureSasCredential]] = None,
```
Comment (seankane-msft) [inline sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py:43]: `AzureSasCredential` or `NamedKeyCredential` right?

Comment (annatisch) [reply sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py:43]: Right - but the work to support NamedKeyCredential is outside the scope of this PR - so just leaving the typehint for what's supported today for now :)

**Review by seankane-msft — APPROVED** [review]
LGTM :shipit:

**Pull request merged**